llvm/lib/Transforms/Vectorize/VPlanRecipes.cpp

```diff
@@ -2563,6 +2495,30 @@ InstructionCost VPReductionRecipe::computeCost(ElementCount VF,
   std::optional<FastMathFlags> OptionalFMF =
       ElementTy->isFloatingPointTy() ? std::make_optional(FMFs) : std::nullopt;
 
+  if (isPartialReduction()) {
+    using namespace llvm::VPlanPatternMatch;
+    VPValue *Mul = getVecOp();
+    // Some chained partial reductions used for complex numbers will have a
+    // negation between the mul and reduction. This extracts the mul from that
+    // pattern to use it for further checking.
+    match(Mul, m_Binary<Instruction::Sub>(m_SpecificInt(0), m_VPValue(Mul)));
+    if (match(Mul,
+              m_Mul(m_ZExtOrSExt(m_VPValue()), m_ZExtOrSExt(m_VPValue())))) {
```

Comment:
The case you're trying to match is the case already implemented by the `VPMulAccumulateReductionRecipe`, so there's no need to try and match this again. For this default case, the call to `getPartialReductionCost` should assume no extends and no opcode.